tests/unit/apps/test_auxiliary.py

```diff
@@ -395,21 +395,7 @@ def test_rabbitmq_server_upgrade_plan_ussuri_to_victoria_auto_restart_False(mode
             description=f"Refresh '{app.name}' to the latest revision of '3.9/stable'",
             coro=model.upgrade_charm(app.name, "3.9/stable"),
         ),
-    ]
-    upgrade_steps += [
-        PreUpgradeStep(
-            description="Auto restarts is disabled, will"
-            f" execute run-deferred-hooks for unit: '{unit.name}'",
-            coro=model.run_action(unit.name, "run-deferred-hooks", raise_on_failure=True),
-        )
-        for unit in app.units.values()
-    ]
-    upgrade_steps += [
-        PreUpgradeStep(
-            description=f"Wait for up to 2400s for app '{app.name}' to reach the idle state",
-            parallel=False,
-            coro=model.wait_for_active_idle(2400, apps=[app.name]),
-        ),
+        *tuple(app.get_run_deferred_hooks_and_restart_pre_upgrade_step()),
```

Comment:
Non blocker for this PR, but a general note that these tests are not really useful, because they aren't very declarative.  We're using the same method to generate the expected output as to generate the output we're testing. :confused:

Reply:
fixed